detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicClassSpec.kt

```diff
@@ -269,6 +270,21 @@ class UndocumentedPublicClassSpec {
         assertThat(subject.compileAndLint(code)).hasSize(3)
     }
 
+    @Test
+    fun `should not report in public companion class if disabled`() {
+        val code = """
+            /** Some doc */
+            public class PublicClass {
+                public companion object
+            }
```

Comment:
I wonder if the existence of any content matters for this feature?

```suggestion
            public class PublicClass {
                public companion object {
                    public val x: String = ""
                }
            }
```

Reply:
I duplicated the test, one for with and one for without content!

Reply:
I didn't mean we need more tests (although more edge case coverage never hurts), but whether it changes the "ignore" logic in any way if there's content vs just an empty placeholder for extensions? Do your use cases apply to all unnamed?

Reply:
What are you referring to by "all unnamed?" Apologies, I'm not quite following 😅

Reply:
all unnamed including body or not.

I see 4 categories of companion objects (syntactically):
* unnamed marker (`companion object;`) - for extensions
* unnamed default (`companion object {}`) - with val's and funs inside = "static" stuff
* named default (`companion object Companion ...`) - not sure why, but this feels like an edge case
* named (`companion object Foo`)

So my thinking is that not all of these have the same rules. For example a named companion object (4th bullet) should definitely always have docs (at minimum to explain the name change). And my original question was whether 1st and 2nd have the same rule regarding docs and the exemption you're adding.

Reply:
> all unnamed including body or not.
> 
> I see 4 categories of companion objects (syntactically):
> * unnamed marker (`companion object;`) - for extensions
> * unnamed default (`companion object {}`) - with val's and funs inside = "static" stuff

I'd argue these are the same class of usages. I think both should be covered under the "unnamed" exception as currently defined. I don't think there's much of a difference in usages. Anecdotally, I often see both usages mixing over time. What may start as an extension only usage, can morph into some funs and vals!

> * named default (`companion object Companion ...`) - not sure why, but this feels like an edge case

This definitely feels like an edge case 😅 

> * named (`companion object Foo`)

This would not be covered by the exemption (hence unnamed!), and it feels like we can let the edge case fall into this one. 

All in, I think we've covered our cases! But let me know if you think differently. :+1:

Reply:
Makes sense! Thanks for clarifying questions and thoughts.